### DISCUSSION TITLE
Unable to get client usage with manual pings and pongs to work

### DISCUSSION THREAD

**Discussion by stuartwallace-valstro** [enisdenjo/graphql-ws#444] [Q&A]
Hello,

I'm attempting to follow the "Client usage with manual pings and pongs". I'm having some problems because the "ping" that I've implemented never gets called. I've set `keepAlive` to 10s and I can see pings being sent but I see none of the logging nor the payload I've added.

Looking at the code, this seems like it's expected. Do I need to implement the timer to send the ping as well?

Thanks!

**Comment by enisdenjo** [comment]
Hey there, thanks for writing. The ["Client usage with manual pings and pongs" recipe](https://github.com/enisdenjo/graphql-ws#custom-client-pinger) is manual issuing of pings/pongs from the client. You have to manually send a ping to the server (to which the server will automatically respond with a pong).

But, honestly, I am a bit confused about this. Would you be so kind and share a repro or the setup? Thanks!
